MapleServer2/Servers/Game/DungeonManager.cs

```diff
@@ -76,17 +76,20 @@ public DungeonSession GetDungeonSessionByInstanceId(int instanceId)
 
         public bool IsDungeonUsingFieldInstance(DungeonSession dungeonSession, FieldManager fieldManager, Player player) //alternatively this could be: IsFieldInstanceUsed in FieldManagerFactory
         {
-            //fieldmanager is the map the player is coming from/that is to be released
             if (dungeonSession == null) //is not null after entering dungeon via directory
             {
-                return false; //no dungeonsession = the map is unused by dungeon, 
+                return false; //no dungeonsession -> the map is unused by dungeon
             }
             //fieldManager.MapId: left map that is to be destroyed
             //player.MapId: travel destination of the player
             //check map that is left: 
-            if (dungeonSession.IsDungeonMap(fieldManager.MapId))  //left map is a dungeon map
+            if (!dungeonSession.IsDungeonMap(fieldManager.MapId)) //left map is not dungeon map e.g. tria
             {
-                //check map the player will travel to: lobby to dungeon and dungeon to lobby
+                return false;
+            }
+            else //left map is a dungeon map
```

Comment:
Since line 88 is returning false, you can remove this else block entirely

Reply:
Do you perhaps mean the else block starting at line 97? 
I could see that the logic in 97+ could be in FieldManagerFactory, but the else at 90+ contains logic that should only be executed in that else.
Am I missing something?